Client/ApiClient.php

```diff
@@ -127,24 +128,33 @@ protected function getResponseData(
         Request $request
     ): ?string {
         $this->logger->debug(
-            "API Request:{$request->getMethod()} {$request->getUri()}",
+            "API Request",
             [
-                'body' => $request->getBody(),
+                'method' => $request->getMethod(),
+                'uri' => $request->getUri(),
+                'body' => (string)$request->getBody(),
             ]
         );
         try {
             $response = $this->client->sendRequest($request);
         } catch (HttpException $e) {
-            throw ApiRequestException::create((string) $request->getUri(), $e);
+            throw ApiRequestException::create((string)$request->getUri(), $e);
         }
-        $body = (string) $response->getBody();
+        $body = (string)$response->getBody();
         $this->logger->debug(
-            "API Response:{$request->getMethod()} {$request->getUri()}",
+            "API Response",
             [
+                'method' => $request->getMethod(),
+                'uri' => $request->getUri(),
+                'status_code' => $response->getStatusCode(),
                 'body' => $body,
             ]
         );
 
+        if ($response->getStatusCode() >= 400) {
```

Comment:
What? This seems weird, there are a lot of codes that are invalid beneath 400

Reply:
From a user point of view maybe but with pure HTTP, errors are 4xx and 5xx (see https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)

Maybe we could improve this, but it was a simple solution to detect most direct errors. I still think there's a problem with this API because on the result we lost a lot of thing (status code, http headers, ...) and the denormalizer do not always fails on invalid properties (maybe with some configuration ? but I my case I had an object with nullified fields because the response did not match what I expected...)

Reply:
OK!